### DISCUSSION TITLE
Change data type of coordinates

### DISCUSSION THREAD

**Discussion by maawoo** [pydata/xarray#5209] [Q&A] [answered]
Hi there! 
I couldn't find a simple answer after searching around, but maybe I'll find it here by asking you...
Is there a simple way of converting the data type of coordinates?
The `x` and `y` coordinates of my dataset are `float64`, which is not really necessary for precision as they're in meters and not degree, and it's kinda inconvenient for plotting. So what's the best way of converting the coordinates to `int32` in this case?

**Comment by keewis** [comment, marked as answer]
unfortunately the canonical method (`astype`) indeed does not allow converting the coords. For data variables it would have been possible to specify a dict that maps names to dtypes (this is missing from the `Dataset.astype` documentation). To work around that, we can use `assign`:

```python
In [8]: ds = xr.Dataset({"a": ("x", [1.3, 6.2])}, coords={"x": [0, 1]})
   ...: ds
Out[8]: 
<xarray.Dataset>
Dimensions:  (x: 2)
Coordinates:
  * x        (x) int64 0 1
Data variables:
    a        (x) float64 1.3 6.2

In [9]: ds.assign(x=lambda ds: ds.x.astype(float))
Out[9]: 
<xarray.Dataset>
Dimensions:  (x: 2)
Coordinates:
  * x        (x) float64 0.0 1.0
Data variables:
    a        (x) float64 1.3 6.2
```

I think extending `astype` to also handle coords would be a welcome addition, but since it currently uses `apply_ufunc` that might be tricky.

**Reply by maawoo** [reply]
Great, thanks for your quick answer! 
A simple 'hack' like this is what I was looking for 👍🏻
